cmd/observe/flows.go

```diff
@@ -136,6 +136,28 @@ func getFlowFiltersYAML(req *observer.GetFlowsRequest) (string, error) {
 	return string(out), nil
 }
 
+// GetHubbleClientFunc is primarily used to mock out the hubble client in some unit tests.
+var GetHubbleClientFunc = func(ctx context.Context, vp *viper.Viper) (client observer.ObserverClient, cleanup func() error, err error) {
+	fi, err := os.Stdin.Stat()
+	if err != nil {
+		return nil, nil, err
+	}
+	if fi.Mode()&os.ModeCharDevice == 0 {
+		// read flows from stdin
+		client = NewIOReaderObserver(os.Stdin)
+		logger.Logger.Debug("Reading flows from stdin")
+	} else {
```

Comment:
Rather than using an `else` close, I'd return in the `if` clause above. This will make the code easier to read and would allow reviewers to more easily catch that `cleanup` is not setup in `stdin` mode, leading to a panic :wink:

Reply:
Ah good point.